menu-api/Repositories/MenuItem_IngredientRepository.cs

```diff
@@ -15,29 +16,63 @@ public MenuItem_IngredientRepository(MenuContext context)
         //Add ingredient to menuItem
         public async Task AddIngredient(MenuItem_Ingredient menuItem_Ingredient)
         {
-            await context.MenuItem_Ingredients.AddAsync(menuItem_Ingredient);
-            await context.SaveChangesAsync();
+            var ingredient = await context.Ingredients.FindAsync(menuItem_Ingredient.IngredientId);
+            var menuItem = await context.MenuItems.FindAsync(menuItem_Ingredient.MenuItemId);
+
+            var _menuItem_Ingredient = await context.MenuItem_Ingredients.FindAsync(menuItem_Ingredient.MenuItemId, menuItem_Ingredient.IngredientId);
+
+            if (_menuItem_Ingredient == null && ingredient != null && menuItem !=null)
```

Comment:
Refactor this code so that it first checks if exceptions must be thrown, and then add the ingredient if everything is OK. Then there is no need for a lot of else if's. I think the github comments will also be resolved this way.